From ab39f46c3a2be807136dd3c3afec5c2e24252fbd Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Wed, 22 Sep 2021 15:23:41 +0300
Subject: [PATCH 07/24] s32g: bl2: Add static checks for BL31/BL32/BL33
 overlaps

Added static checks to prevent the misconfiguration of
the BL31, BL32 and BL33 memory ranges such that they
would overlap.

Issue: ALB-7733
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32g/s32g_bl2_el3.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/plat/nxp/s32g/s32g_bl2_el3.c b/plat/nxp/s32g/s32g_bl2_el3.c
index d1f582190..d391c6e12 100644
--- a/plat/nxp/s32g/s32g_bl2_el3.c
+++ b/plat/nxp/s32g/s32g_bl2_el3.c
@@ -562,6 +562,12 @@ void s32g_el3_mmu_fixup(void)
 	};
 	int i;
 
+	/* Check the BL31/BL32/BL33 memory ranges for overlapping */
+	_Static_assert(S32G_BL32_BASE + S32G_BL32_SIZE <= BL33_BASE,
+				"BL32 and BL33 memory ranges overlap!");
+	_Static_assert(BL33_BASE + S32G_BL33_IMAGE_SIZE <= BL31_BASE,
+				"BL33 and BL31 memory ranges overlap!");
+
 	/* The calls to mmap_add_region() consume mmap regions,
 	 * so they must be counted in the static asserts
 	 */
-- 
2.17.1

